include/fiction/algorithms/physical_design/orthogonal.hpp

```diff
@@ -534,46 +629,56 @@ class orthogonal_impl
                         node2pos[n] = connect_and_place(layout, t, ctn.color_ntk, n, pre1_t, pre2_t, fc.constant_fanin);
                     }
 
-                    // create PO at applicable position
                     if (ctn.color_ntk.is_po(n))
                     {
-                        const auto n_s = node2pos[n];
-
-                        tile<Lyt> po_tile{};
-
-                        // determine PO orientation
-                        if (is_eastern_po_orientation_available(ctn, n))
+                        if (place_outputs_last)
                         {
-                            po_tile = layout.east(static_cast<tile<Lyt>>(n_s));
-                            ++latest_pos.x;
+                            if (!is_eastern_po_orientation_available(ctn, n) ||
+                                std::find(multi_out_nodes.begin(), multi_out_nodes.end(), n) != multi_out_nodes.end())
+                            {
+                                ++latest_pos.y;
+                            }
                         }
                         else
                         {
-                            po_tile = layout.south(static_cast<tile<Lyt>>(n_s));
-                            ++latest_pos.y;
-                        }
+                            const auto n_s = node2pos[n];
 
-                        // check if PO position is located at the border
-                        if (layout.is_at_eastern_border(po_tile))
-                        {
-                            layout.create_po(n_s,
-                                             ctn.color_ntk.has_output_name(po_counter) ?
-                                                 ctn.color_ntk.get_output_name(po_counter++) :
-                                                 fmt::format("po{}", po_counter++),
-                                             po_tile);
-                        }
-                        // place PO at the border and connect it by wire segments
-                        else
-                        {
-                            const auto anker = layout.create_buf(n_s, po_tile);
+                            tile<Lyt> po_tile{};
+
+                            // determine PO orientation
+                            if (is_eastern_po_orientation_available(ctn, n))
+                            {
+                                po_tile = layout.east(static_cast<tile<Lyt>>(n_s));
+                                ++latest_pos.x;
```

Comment:
Instead of manually incrementing x and y positions, rather use `east()` and `south()`